feat: add release.yml for 2.3.2 release

metacosm opened the pull request:
Fixes #115

This file is meant to record the components that are needed for a given release as well as timeline information. Each component definition will be used to create a JIRA ticket to inform stakeholders that we're planning on releasing a new Spring Boot version and that action should be taken by validating the associated list of artifacts on the specified Spring Boot release. 

The association between component and artifacts is done via the component's associated properties so we need to make sure that these properties are correct.

Similarly, we need to make sure that the proper JIRA project is used to create the associated tickets.

geoand commented:
This is probably a stupid question, but where does one see the exact SB version the release is for?

metacosm commented:
> This is probably a stupid question, but where does one see the exact SB version the release is for?

This is not a stupid question at all! 😄 

The version is derived from the pom.xml that exists in the same repository.

geoand commented:
Gotcha.

But isn't the intent of the file to provide all the information that one needs to know without having to look into multiple places?

metacosm commented:
> Gotcha.
> 
> But isn't the intent of the file to provide all the information that one needs to know without having to look into multiple places?

It's also to avoid duplicated information but I definitely could add an explicit version that could also add some extra layer of validation…

metacosm commented:
> Same remark as Georgios, it should help to see the SB version.
> I'm missing these ones:
> 
>     * spring-cloud-kubernetes
>     * cxf-spring-boot-starter-jaxrs
>     * vertx-spring-boot
>       Are not they there for any reason?

The role of components is to record which dependencies we have for each Red Hat product that we depend on. SCK is not one of them so it's expected that it's not recorded there.

CXF is not bundled anymore since 2.2.

Vert.x is defined here: https://github.com/snowdrop/spring-boot-bom/pull/117/files#diff-825f26e19a91d7ebe39342065295d457R73-R77

Am I missing something?

aureamunoz commented:
Ok, understood, thanks.

How did you figured out the listed CVEs?

metacosm commented:
> Ok, understood, thanks.
> 
> How did you figured out the listed CVEs?

By going through them with @zekaguilar and deciding which should be assigned to this release or not.

jacobdotcosta commented:
The entries regarding Narayana, Vert.X and Dekorate are for us to decide and document which version of these components will be used, right?

jacobdotcosta commented:
> > Gotcha.
> > But isn't the intent of the file to provide all the information that one needs to know without having to look into multiple places?
> 
> It's also to avoid duplicated information but I definitely could add an explicit version that could also add some extra layer of validation…

I'm not sure about this duplication, because then we have to automate the update of this file also, and also check it again. I thought of this file as just a proxy for the creation of the tickets using the `pom.xml` in the same release.

metacosm commented:
> > > Gotcha.
> > > But isn't the intent of the file to provide all the information that one needs to know without having to look into multiple places?
> > 
> > 
> > It's also to avoid duplicated information but I definitely could add an explicit version that could also add some extra layer of validation…
> 
> I'm not sure about this duplication, because then we have to automate the update of this file also, and also check it again. I thought of this file as just a proxy for the creation of the tickets using the `pom.xml` in the same release.

Actually, this file shouldn't be created automatically since it's the result of a human decision so it makes sense to have that "duplicated" information to also check that it's actually put in the right spot with a `pom.xml` that matches the expectations set in `release.yml` (i.e. it should be an error to have a mismatch between the version that's set in `release.yml` and `pom.xml`).

gytis commented on release.yml:
```diff
@@ -0,0 +1,92 @@
+components:
+  # List of JIRA Projects contributing to a Snowdrop release
+  # They will be used to create a component or starter
+  - name: Hibernate / Hibernate Validator / Undertow
```
Does anybody know how should the Narayana and Vert.x versions be handled? We include the starters but do not include the underlying component versions in the same way as we do Hibernate, Infinispan etc. Is it OK since we do the starter prod ourselves or is this an issue?

metacosm replied:
Good point. We probably need to create components for these dependencies as well since we need the product team to tell us which version we're supposed to be using. Do you have the associated JIRA project information for these products?

gytis replied:
JBTM for Narayana and I don't know for Vert.x.
Do they need to tell us a productised/upstream version that we should use or a version that they guarantee to work with Spring Boot (because the later they will not know).

metacosm replied:
If I'm correct in the assumption that we're the ones making sure that a given version works with a given Spring Boot version, they only need to tell us which productized version we should use.

gytis replied:
Yes, we're the ones maintaining these starters and the project teams are not involved in Spring Boot integration.

metacosm replied:
So should we create a ticket for these products as well to ask them which productized version we're supposed to use? Or are we going to handle this manually?

gytis replied:
So far we've been doing it manually without issues but one exception (which was resolved and shouldn't happen again). I don't think there is much need for it.
However, if the idea of this process change is to create issues for all the third party components and provide a trace record, it might make sense. In such case the issue might have to be different from the other projects though.

cmoulliard commented:
> SCK is not one of them so it's expected that it's not recorded there.

As we ship it, then it must be tracked as we do also for AMQP, opentracing, infinispan, vertx, ...

metacosm commented:
> > SCK is not one of them so it's expected that it's not recorded there.
> 
> As we ship it, then it must be tracked as we do also for AMQP, opentracing, infinispan, vertx, ...

We could certainly track it but adding it to the `release.yml` won't have any effect since we won't create a ticket for it.

cmoulliard commented:
> We could certainly track it but adding it to the `release.yml` won't have any effect since we won't create a ticket for it.

From my point o view a ticket should be created and assigned to Snowdrop team as we lead/manage: SCK, Narayana Spring starter, ....

metacosm commented:
@cmoulliard @gytis I've simplified the model a little bit and added `product` fields to the starters we own so that when the tool creates a ticket under `ENTSBT` to record that work is needed on these starters, we also create a ticket under the associated product JIRA to request information on which supported version we should be using in our starter.

cmoulliard commented on release.yml:
```diff
@@ -0,0 +1,92 @@
+    properties:
+      - narayana-starter
+
+  - name: Vert.X
```
Vert.X should be Vert.x

cmoulliard commented on release.yml:
```diff
@@ -0,0 +1,92 @@
+      project: ENTSBT
+    properties:
+      - dekorate
+
```
What about SCK ?

metacosm replied:
It's right above: https://github.com/snowdrop/spring-boot-bom/pull/117/files#diff-825f26e19a91d7ebe39342065295d457R74-R77

cmoulliard replied:
Perfect

gytis commented on release.yml:
```diff
@@ -0,0 +1,89 @@
+  - name: RESTEasy
+    jira: RESTEASY
+    properties:
+      - resteasy
```
I might be forgetting something but why do we need to specify a RESTEasy version when we already specify a RESTEasy starter? Wouldn't the starter bring all that is needed?

gytis replied:
This question is not meant to block this PR. It is rather a more generic question about the dependency included in the BOM.

metacosm replied:
Theoretically, we don't need them but I suspect that we needed them to override the RESTEasy version from the starter at some point. Don't recall for sure. See: 40f61485bbcead43b7ed781e1fa885b0db7b6433

cmoulliard commented:
> If we move forward with adding the upstream version explicitly to this file, which I'm 👍 with, how would the file result?

Why would you like to do that ? ATM the code of this project do not handle such versions. So what is then the purpose of adding upstream version ...

jacobdotcosta commented:
> > If we move forward with adding the upstream version explicitly to this file, which I'm +1 with, how would the file result?
> 
> Why would you like to do that ? ATM the code of this project do not handle such versions. So what is then the purpose of adding upstream version ...

It was commented above as a possibility, or was it only for the spring-boot version?

metacosm commented:
> > > If we move forward with adding the upstream version explicitly to this file, which I'm +1 with, how would the file result?
> > 
> > 
> > Why would you like to do that ? ATM the code of this project do not handle such versions. So what is then the purpose of adding upstream version ...
> 
> It was commented above as a possibility, or was it only for the spring-boot version?

The generated description (which should be fairly easy to infer if you look at the template) looks like for the Hibernate case:

```markdown
The Snowdrop team is really pleased to inform you that we're planning to release a new Snowdrop BOM on *14 Sep 2020*
based on Spring Boot *2.3.2*.

We, however, cannot release a new version without your support as Red Hat's Spring Boot stack relies on supported versions for the following artifact(s)
that are currently owned by your project:
    * org.hibernate:hibernate-core (upstream version is 5.4.18.Final)
    * org.hibernate:hibernate-entitymanager (upstream version is 5.4.18.Final)
    * org.hibernate.validator:hibernate-validator (upstream version is 6.1.5.Final)
    * io.undertow:undertow-core (upstream version is 2.1.3.Final)
    * io.undertow:undertow-servlet (upstream version is 2.1.3.Final)
    * io.undertow:undertow-websockets-jsr (upstream version is 2.1.3.Final)


We expect to start testing our new release based on Spring Boot *2.3.2* on *18 Aug 2020*.

We will support this new release until *30 Sep 2021* and therefore require compatibility and
support statement from your team for this Spring Boot version in the form of the following information:
- Product name and version (e.g. JWS - 5.3.1)
- Date of end of support
- Supported artifact GAV (if changed) or simply version (if name hasn't changed) (e.g. org.apache.tomcat.embed:tomcat-embed-core:9.0.30-redhat-00001)
- PNC/Indy URL (e.g. [http://indy.psi.redhat.com/browse/maven/]...)
- MRRC URL, if already released (e.g. [https://maven.repository.redhat.com/ga/]...)

Note that we don't require a new release of your product as long as the artifacts you provide us with are tested and
supported on Spring Boot *2.3.2* until *30 Sep 2021*.

Thank you in advance for your help!
```

cmoulliard commented:
LGTM. Can be merged and JIR tickets created. Do you agree @zekaguilar ?

jacobdotcosta commented:
LGTM

metacosm commented:
Note also that in the case of starters that we own, we will create a ticket for the starter in our own JIRA which will be linked to a ticket in the associated product JIRA.
This is how that associated product JIRA will look like:
```markdown
The Snowdrop team is really pleased to inform you that we're planning to release a new Snowdrop BOM on *14 Sep 2020*
based on Spring Boot *2.3.2*.

Some of our artifacts depend on your product and would, therefore, need to know which product version we should be
using for this new release which we intend to support until *30 Sep 2021*.


We expect to start testing our new release based on Spring Boot *2.3.2* on *18 Aug 2020*.
We would therefore request a support statement from your team in the form of the following information:
- Product name and version (e.g. JWS - 5.3.1)
- Supported version (e.g. 9.0.30-redhat-00001)
- Date of end of support
- PNC/Indy URL (e.g. [http://indy.psi.redhat.com/browse/maven/]...)
- MRRC URL, if already released (e.g. [https://maven.repository.redhat.com/ga/]...)

Please let us know if the product support will lapse before *30 Sep 2021*.

Thank you in advance for your help!
```

zekaguilar commented:
Agree !
